msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -180,4 +440,34 @@ private MultiTenantAccount getAccountFromICacheRecordList(@NonNull List<ICacheRe
 
         return (MultiTenantAccount) account.get(0);
     }
+
+    @Override
+    public void acquireTokenSilentAsync(@NonNull final String[] scopes,
+                                        @NonNull final AuthenticationCallback callback) {
+        acquireTokenSilent(
+                scopes,
+                getPersistedCurrentAccount(),
```

Comment:
Are we planning to support ADAL - > Single Account MSAL migration?

Reply:
If yes..

For non-FLW, single account mode, I'm expecting that the user should not have to call getCurrentAccount() at all, so we could try to do one-time getCurrentAccount() here - for example.

Reply:
Let's discuss this one in person

Reply:
I would expect `getCurrentAccount()` to be called when broker is in-play (as, in the case of FoCI, another app could remove the account) -- since apps won't know that the broker is necessarily being used or not, I would say that developers ought to call `getCurrentAccount()` before trying to authenticate a user